diff --git a/field_formatter_label.module b/field_formatter_label.module
index 134bb70..8d3a68f 100644
--- a/field_formatter_label.module
+++ b/field_formatter_label.module
@@ -57,8 +57,8 @@ function field_formatter_label_module_implements_alter(&$implementations, $hook)
 }
 
 /**
-*  Implementation of hook_theme_registry_alter
-*/
+ * Implements hook_theme_registry_alter().
+ */
 function field_formatter_label_theme_registry_alter(&$theme_registry) {
   // Move our hook_preprocess_field to the end of the array.
   $theme_registry['field']['preprocess functions'] = array_diff($theme_registry['field']['preprocess functions'], array('field_display_label_preprocess_field'));
@@ -66,11 +66,14 @@ function field_formatter_label_theme_registry_alter(&$theme_registry) {
 }
 
 /**
-*  Preprocess function to replace the regular label with the display label
-*/
+ * Preprocess function to replace the regular label with the display label.
+ */
 function field_formatter_label_preprocess_field(&$variables) {
+  $view_mode = $variables['element']['#view_mode'];
   $field = field_info_instance($variables['element']['#entity_type'], $variables['element']['#field_name'], $variables['element']['#bundle']);
-  if (isset($field['display_label']) && strlen(trim($field['display_label'])) > 0) {
-    $variables['label'] = $field['display_label'];
+  $settings = isset($field['display'][$view_mode]['settings']) ? $field['display'][$view_mode]['settings'] : $field['display']['default']['settings'];
+  if ($variables['element']['#label_display'] != 'hidden' &&
+    strlen(trim($settings['field_formatter_label'])) > 0) {
+    $variables['label'] = t(check_plain($settings['field_formatter_label']));
   }
 }
