diff --git a/simplesamlphp_auth.module b/simplesamlphp_auth.module
index 74c12d6..b3698f2 100644
--- a/simplesamlphp_auth.module
+++ b/simplesamlphp_auth.module
@@ -365,7 +365,7 @@ function simplesamlphp_auth_user_insert(&$edit, $account, $category = NULL) {
 
   if ($category = 'account') {
     // If user registration has a valid session...
-    if ($_simplesamlphp_auth_as->isAuthenticated()) {
+    if ($_simplesamlphp_auth_as->isAuthenticated() && _simplesamlphp_auth_get_authname() == $account->init) {
       // Get name from default attributes.
       try {
         _simplesaml_auth_debug(t('Registering user [%acctname]', array('%acctname' => $account->name)));
